avocado/core/messages.py

```diff
@@ -153,10 +153,37 @@ class BaseRunningMessageHandler(BaseMessageHandler):
     """Base interface for resolving running messages."""
 
     @staticmethod
-    def _save_message_to_file(filename, buff, task, mode='a'):
+    def _save_message_to_file(filename, buff, task, encoding=None):
+        """
+        Method for saving messages into the file
+
+        It can decode and save messages.The message will be decoded when
+        encoding is not None. When the decoded message doesn't end with a new
+        line the new line will be added. Every message is saved in the append
+        mode.
+
+        :param filename: name of the file
+        :type filename: str
+        :param buff: message to be saved
+        :type buff: bytes
+        :param task: message related task.
+        :type task: :class:`avocado.core.nrunner.Task`
+        :param encoding: encoding of buff, default is None
+        :type encoding: str
+        """
+
+        def _save_to_file(file_name, mode):
+            with open(file_name, mode) as fp:
+                fp.write(buff)
+
         file = os.path.join(task.metadata['task_path'], filename)
-        with open(file, mode) as fp:
-            fp.write(buff)
+        if encoding:
+            buff = buff.decode(encoding)
+            if not buff.endswith("\n"):
+                buff = "%s\n" % buff
```

Comment:
My only concern here is this new line only when encoding is used. If we use binary mode no new line is added.

It feels to me that this should not be part of this method (not following the Single Responsibility Principle), but I might be wrong here.

Could you refresh my mind, please, why is this being added?

Reply:
I can put it into a different method.
The reason for adding new line is when the message is string we understand each message as a line  or set of lines which have to ends with new line character.

Reply:
My suggestion then, is to move this logic to outside of this method to keep the SRP.

Reply:
I agree, already did the change.